### PR TITLE
BUGFIX: #3624 Node::getProperty does not always return list for references

### PR DESCRIPTION
**Pull request by mhsdesign**
EEL can only operate on an array of nodes if the `[0]` item is a node (simple duck typing)

Instead of fixing eel like proposed here https://github.com/neos/neos-development-collection/pull/3946 with this fix we avoid returning non 0 indexed arrays here:

```
${q(node).property("someReference")}
```

Currently, this might indeed return an array with holes like `[1 => NODE, 2 => NODE, 5 => NODE]` if the identifiers in fields 0, 3 and 4 are not resolvable.

Thats because of the "unsafe" `array_filter` method in `resolvePropertyReferences` https://github.com/neos/neos-development-collection/blob/378a029d0cc7ea6acb853751e7592873584a4aac/Neos.ContentRepository/Classes/Domain/Model/Node.php#L961 which will leave holes in the array.

Using `array_filter` was introduced with Neos 2.2 so this is technically a regression of https://github.com/neos/neos-development-collection/commit/87804e12082e7d6d06bd22f50739e20eeaa45539 ^^

Fixes: #3624

### CODE REVIEW

**Comment by bwaidelich** [comment]
This one begs for a corresponding test

File: Neos.ContentRepository/Classes/Domain/Model/Node.php [hunk 1]
```diff
@@ -960,11 +960,15 @@ public function getProperty($propertyName, bool $returnNodesAsIdentifiers = fals
...
-        return array_filter($nodes);
+        $nodes = [];
+        foreach ($value as $nodeIdentifier) {
+            $node = $this->context->getNodeByIdentifier($nodeIdentifier);
+            // $node can be NULL if the node is not visible (or removed) according to the current content context:
+            if ($node !== null) {
+                $nodes[] = $node;
+            }
+        }
+        return $nodes;
```
Comment (bwaidelich) [inline Neos.ContentRepository/Classes/Domain/Model/Node.php:971]: Just adding an `array_values()` around the return would work, as well:
```suggestion
        $nodes = array_map(function ($nodeIdentifier) {
            return $this->context->getNodeByIdentifier($nodeIdentifier);
        }, $value);

        return array_values(array_filter($nodes));
```

**Review by bwaidelich — APPROVED** [review]
I took the freedom to add a unit test.
Also suggested a slightly different approach, but that's just a FYI really

File: Neos.ContentRepository/Classes/Domain/Model/Node.php (hunk 1 above)
Comment (mhsdesign) [reply Neos.ContentRepository/Classes/Domain/Model/Node.php:971]: yes thanks i know ;) But found it more funny to reuse the 8 year old original pattern - which also seems more intuitive ^^

**Review by dlubitz — APPROVED** [review]

**Pull request merged**